components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/datasource/QueryExecutor.java

```diff
@@ -89,12 +86,13 @@ public boolean executeDeleteQuery(String uuid) throws BusinessRulesDatasourceExc
         PreparedStatement statement = null;
         try {
             conn = dataSource.getConnection();
-            conn.setAutoCommit(true);
+            conn.setAutoCommit(false);
             statement = getDeleteQuery(conn, uuid);
             if (statement == null) {
                 return false;
             }
             result = statement.execute();
+            conn.commit();
             return result;
         } catch (SQLException e) {
             log.error("Deleting business rule with uuid '" + uuid + " is failed due to " + e.getMessage());
```

Comment:
don't we need the stack trace here ?

Reply:
Fixed by 2c0119c4c48e1077512e685493252c388e68a520